src/kudu/consensus/raft_consensus.cc

```diff
@@ -2389,10 +2390,19 @@ Status RaftConsensus::CheckBulkConfigChangeAndGetNewConfigUnlocked(
             return Status::InvalidArgument("peer must have last_known_addr specified",
                                            SecureShortDebugString(req));
           }
-          if (peer.member_type() != RaftPeerPB::VOTER && peer.has_attrs() &&
-              peer.attrs().has_quorum_id() && peer.attrs().quorum_id() != "") {
-            return Status::InvalidArgument("Non-voter must not have quorum_id",
+          if (peer.has_member_type() && peer.member_type() == RaftPeerPB::VOTER) {
+            if (FLAGS_enable_flexi_raft && IsUseQuorumId(committed_config.commit_rule())) {
+              if (!PeerHasNonEmptyQuorumId(peer)) {
+                return Status::InvalidArgument("Peer must have a non-empty quorum_id in "
+                                                "QuorumId type quorum",
+                                                SecureShortDebugString(req));
+              }
+            }
+          } else {
```

Comment:
nit: else if {

Reply:
Why do we add "if"?

Above "if" is check whether it is VOTER. This else means it is "NON-VOTER".

Reply:
Oh I see what you mean. Above if condition is VOTER, and else means "NON-VOTER". I feel like nested condition makes more sense in this case.